thriftclient/client.go

```diff
@@ -0,0 +1,67 @@
+package thriftclient
+
+import (
+	"context"
+	"errors"
+
+	"github.com/apache/thrift/lib/go/thrift"
+
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+var (
+	// ErrNilTClient is returned when a nil thrift.TClient is passed to
+	// NewMonitoredClient.
+	ErrNilTClient = errors.New("recieved a nil thrift client")
+)
+
+// MonitoredClient implements the thrift.TClient interface and can be used to
+// wrap thrift calls in a client span.
+//
+// Calls will only be wrapped in client spans if the given context has a parent
+// span (either a server or "active" local/client span) set on it.  If no such
+// span is set, then the call will still proceed, it will just not be monitored
+// by a client span.
+type MonitoredClient struct {
+	client thrift.TClient
+}
+
+// NewMonitoredClient returns a pointer to a new MonitoredClient that wraps the
+// given thrift.TClient calls in thrift client spans.
+//
+// If c is nil, will return a nil client and an error rather than panicking.
+func NewMonitoredClient(c thrift.TClient) (*MonitoredClient, error) {
+	if c == nil {
+		return nil, ErrNilTClient
```

Comment:
Would it actually be better to just panic here rather than adding an error to the return type?  That doesn't seem like an uncommon approach to this sort of thing.

Reply:
just let it panic should be fine.